src/us/kbase/workspace/test/controllers/handle/HandleServiceController.java

```diff
@@ -35,115 +34,118 @@
-		2014/07/27 23:26:42 15811 reading config from /kb/deployment/deployment.cfg
-		2014/07/27 23:26:42 15811 using http://localhost:7044 as the default shock server
-		{"attribute_indexes":[""],"contact":"shock-admin@kbase.us","documentation":"http://localhost:7044/wiki/","id":"Shock","resources":["node"],"type":"Shock","url":"http://localhost:7044/","version":"0.8.16"}DBI connect('hsi;host=localhost','hsi',...) failed: Access denied for user 'hsi'@'localhost' (using password: YES) at /kb/deployment/lib/Bio/KBase/AbstractHandle/AbstractHandleImpl.pm line 67
-		Cannot read config file /etc/log/log.conf at /kb/deployment/lib/Bio/KBase/Log.pm line 282.
-		HTTP::Server::PSGI: Accepting connections at http://0:5000/
-		
-		--port for port
-		 */
-		
-		ProcessBuilder handlepb = new ProcessBuilder(plackupExe, "--port",
-				"" + handleServicePort, abstractHandlePSGIpath)
+		File hsIniFile = createHandleServiceDeployCfg(mongo, shockHost, authURL,
+				shockAdminToken, handleAdminRole);
+		String lib_dir = "lib";
+		downloadSourceFiles(tempDir, lib_dir);
+
+		String lib_dir_path = tempDir.resolve(lib_dir).toAbsolutePath().toString();
+		ProcessBuilder handlepb = new ProcessBuilder("uwsgi", "--http", 
+				":" + handleServicePort, "--wsgi-file",
+				"AbstractHandle/AbstractHandleServer.py", "--pythonpath", lib_dir_path)
 				.redirectErrorStream(true)
 				.redirectOutput(tempDir.resolve("handle_service.log").toFile());
 		Map<String, String> env = handlepb.environment();
 		env.put("PERL5LIB", perl5lib);
-		env.put("KB_DEPLOYMENT_CONFIG", hsIniFile.toString());
+		env.put("KB_DEPLOYMENT_CONFIG", hsIniFile.getAbsolutePath().toString());
 		env.put("KB_SERVICE_NAME", HANDLE_SERVICE_NAME);
+		env.put("KB_AUTH_TOKEN", shockAdminToken.getToken());
+		env.put("PYTHONPATH", lib_dir_path);
+		handlepb.directory(new File(lib_dir_path));
 		handleService = handlepb.start();
 		
 		Thread.sleep(1000); //let the service start up
+	}
+	
+	private void downloadSourceFiles(Path parentDir, String subDir) throws IOException {
+		// download source files from github repo
 		
-		handleManagerPort = findFreePort();
+		Path lib_root = parentDir.resolve(subDir);
+		Files.createDirectories(lib_root);
 		
-		File hmIniFile = createHandleManagerDeployCfg(
-				shockAdminToken, handleManagerAllowedUser, authServiceURL);
+		Path handle_dir = lib_root.resolve("AbstractHandle");
+		Files.createDirectories(handle_dir);
 		
-		ProcessBuilder handlemgrpb = new ProcessBuilder(plackupExe, "--port",
-				"" + handleManagerPort, handleManagerPSGIpath)
-				.redirectErrorStream(true)
-				.redirectOutput(tempDir.resolve("handle_mngr.log").toFile());
-		env = handlemgrpb.environment();
-		env.put("PERL5LIB", perl5lib);
-		env.put("KB_DEPLOYMENT_CONFIG", hmIniFile.toString());
-		handleManager = handlemgrpb.start();
+		String handle_repo_prefix = "https://raw.githubusercontent.com/kbase/handle_service2/develop/lib/AbstractHandle/";
```

Comment:
I think I would rather leave it up to the user to set up the handle service files so that the handle service version doesn't change during testing unexpectedly, and so that we can test against older versions if we're trying to debug.

I'd just make a helper script or instructions for setting up the handle service appropriately so the tests are runnable, and add a configuration item in test.cfg that points to the proper directory. We may want the user to be responsible for setting up the python path if that's necessary to get virtualenvs to work correctly, not sure about that.

Reply:
👍